File: BugManagement/Models/Bug.cs
namespace BugManagement.Models
{
    public class Bug
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string Status { get; set; } = "Open"; // Open, Work In Progress, Closed, Hold, Rejected
        public string? Severity { get; set; }
        public string? AssignedTo { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public bool IsDeleted { get; set; }
    }
}

File: BugManagement/bug-dashboard/src/app/models/bug.ts
namespace BugManagement.bug_dashboard.src.app.models
{
  // src/app/models/bug.ts
  export interface Bug {
    id?: number;           // optional because new bug doesn't have id yet
    title: string;
    description?: string;
    status: string;        // Open, Work In Progress, Closed, Hold, Rejected
    severity?: string;
    assignedTo?: string;
    createdAt?: string;
    updatedAt?: string;
  }

}
